cpp/cmake/thirdparty/patches/thrust_override.json

```
@@ -0,0 +1,34 @@
+
+{
+  "packages" : {
+    "Thrust" : {
+      "patches" : [
+        {
+          "file" : "Thrust/install_rules.diff",
+          "issue" : "Thrust 1.X installs incorrect files [https://github.com/NVIDIA/thrust/issues/1790]",
+          "fixed_in" : "2.0"
+        },
+        {
+          "file" : "${current_json_dir}/thrust_transform_iter_with_reduce_by_key.diff",
+          "issue" : "Support transform iterator with reduce by key [https://github.com/NVIDIA/thrust/pull/1805]",
+          "fixed_in" : "2.1"
+        },
+        {
+          "file" : "${current_json_dir}/thrust_disable_64bit_dispatching.diff",
+          "issue" : "Remove 64bit dispatching as not needed by libcudf [https://github.com/rapidsai/cudf/pull/11437]",
+          "fixed_in" : ""
+        },
+        {
+          "file" : "${current_json_dir}/thrust_faster_sort_compile_times.diff",
+          "issue" : "Improve Thrust sort compile times [https://github.com/rapidsai/cudf/pull/10577]",
+          "fixed_in" : ""
+        },
+        {
+          "file" : "${current_json_dir}/thrust_faster_scan_compile_times.diff",
+          "issue" : "Improve Thrust scane compile times [https://github.com/rapidsai/cudf/pull/8183]",
```

I don't know what the description for this one should be. Something with policy versions. 🤷‍♂️
```suggestion
          "issue" : "Improve Thrust scan compile times [https://github.com/rapidsai/cudf/pull/8183]",
```

---

Like the dispatch change, this one also helps reduce the number of kernels generated for `scan` and `reduce`. @elstehle

---

I updated the issue string to a better explanation ( and fixed the typo ).